nautobot/dcim/models/sites.py

```diff
@@ -215,3 +215,10 @@ def to_csv(self):
             self.contact_email,
             self.comments,
         )
+
+    def clean_fields(self, exclude=None):
+        if self.longitude is not None:
+            self.longitude = str(self.longitude)
```

Comment:
Is this sufficient or do we need to explicitly specify the target precision when converting to a string? What happens if you try to set `longitude = 0.123456789" for example?

Reply:
Yeah I think there is a 8 digit limit on both of these fields.

Reply:
Raising an error whenever the maximum number of digits is reached makes perfect sense.

Reply:
This is strange to me because it's a `DecimalField` which means the internal value is a `decimal.Decimal` object.

There appears to be some incongruence here in these fields as defined on the model, could this be the culprit?

```python
    latitude = models.DecimalField(
        max_digits=8,
        decimal_places=6,
        blank=True,
        null=True,
        help_text="GPS coordinate (latitude)",
    )
    longitude = models.DecimalField(
        max_digits=9,
        decimal_places=6,
        blank=True,
        null=True,
        help_text="GPS coordinate (longitude)",
    )
```

It should be `8`, but our field has it defined as `9`. https://en.wikipedia.org/wiki/Decimal_degrees#Precision

Reply:
I'd assumed `max_digits` was total digits before and after the decimal, i.e. `123.456789` with 6 decimal places but 9 digits?

Reply:
Ah yes, that vibes. 3 + 6 since that falls under "individual humans" in the table I linked.

Reply:
The real root cause is that a `float` can in fact be cast to a `Decimal`, except that the precision is lost:

```python
In [16]: decimal.Decimal(55.22)
Out[16]: Decimal('55.219999999999998863131622783839702606201171875')

In [17]: decimal.Decimal("55.22")
Out[17]: Decimal('55.22')
```

So when you try to pass a `float` through a `DecimalValidator` (the validator used underneath by `DecimalField`, this happens:

```python
In [20]: from django.core.validators import DecimalValidator

In [21]: validator = DecimalValidator(max_digits=9, decimal_places=6)

In [22]: validator(decimal.Decimal(55.22))
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In [22], line 1
----> 1 validator(decimal.Decimal(55.22))

File ~/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/django/core/validators.py:476, in DecimalValidator.__call__(self, value)
    473 whole_digits = digits - decimals
    475 if self.max_digits is not None and digits > self.max_digits:
--> 476     raise ValidationError(
    477         self.messages['max_digits'],
    478         code='max_digits',
    479         params={'max': self.max_digits, 'value': value},
    480     )
    481 if self.decimal_places is not None and decimals > self.decimal_places:
    482     raise ValidationError(
    483         self.messages['max_decimal_places'],
    484         code='max_decimal_places',
    485         params={'max': self.decimal_places, 'value': value},
    486     )

ValidationError: ['Ensure that there are no more than 9 digits in total.']
```